Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs

```diff
@@ -1,51 +1,132 @@
 ﻿using Microsoft.AspNetCore.Mvc;
-using NBitcoin;
+using Microsoft.Extensions.PlatformAbstractions;
+using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.RPC.Models;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Stratis.Bitcoin.BlockStore;
+using Stratis.Bitcoin.MemoryPool;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.RPC.Controllers
 {
     //TODO: Need to be extensible, should be FullNodeController
-    public partial class ConsensusController : Controller
+    public partial class FullNodeController : Controller
     {
+        private Builder.IFullNode _FullNode;
+        private NodeSettings _Settings;
+        private NBitcoin.Network _Network;
+        private ConsensusValidator _ConsensusValidator;
+        private NBitcoin.ChainBase _Chain;
+        private ChainBehavior.ChainState _ChainState;
+        private BlockStoreManager _BlockManager;
+        private MempoolManager _MempoolManager;
+        private Connection.ConnectionManager _ConnectionManager;
+
+        public FullNodeController(
+            Builder.IFullNode fullNode = null,
+            NodeSettings nodeSettings = null,
+            NBitcoin.Network network = null,
+            Consensus.ConsensusValidator consensusValidator = null,
+            NBitcoin.ConcurrentChain chain = null,
+            ChainBehavior.ChainState chainState = null,
+            BlockStoreManager blockManager = null,
+            MempoolManager mempoolManager = null,
+            Connection.ConnectionManager connectionManager = null)
+        {
+            this._FullNode = fullNode;
+            this._Settings = nodeSettings;
+            this._Network = network;
+            this._ConsensusValidator = consensusValidator;
+            this._Chain = chain;
+            this._ChainState = chainState;
+            this._BlockManager = blockManager;
+            this._MempoolManager = mempoolManager;
+            this._ConnectionManager = connectionManager;
+        }
+
         [ActionName("stop")]
         public Task Stop()
         {
-            _FullNode.Dispose();
+            if (this._FullNode != null)
+            {
+                this._FullNode.Dispose();
+                this._FullNode = null;
+            }
             return Task.CompletedTask;
         }
 
         [ActionName("getrawtransaction")]
         public async Task<TransactionModel> GetRawTransaction(string txid, int verbose = 0)
         {
-            uint256 trxid;
+            NBitcoin.uint256 trxid;
```

Comment:
There was at least one ambiguity between using of NBitcoin, and of Stratis.Bitcoin, so I elected to leave out the NBitcoin namespace for clarity. If prefered I'll using NBitcoin and disambiguate only where needed.